File: service/article.go
package service

import (
	"github.com/makishi00/go-vue-bbs/model"
)

var Article = article{}

type article struct {
}

func (a *article) Store(article model.Article, userId uint) model.Article {
	article.UserID = userId
	db.Create(&article)
	return article
}

func (a *article) Show() []model.Article {
	var article []model.Article
	db.Find(&article)
	return article
}

func (a *article) ExistArticleIdByUserId(id int, userId uint) bool {
	var articles []model.Article
	db.Where("id = ? AND user_id = ?", id, userId).Find(&articles)

	return len(articles) != 0
}

func (a *article) Delete(article model.Article, id int, userId uint) model.Article {
	db.Where("id = ? AND user_id = ?", id, userId).Delete(&article)
	return article
}

func (a *article) Edit(article model.Article, id int, userId uint, title string, body string) model.Article {
	db.Model(&article).Where("id = ? AND user_id = ?", id, userId).Updates(map[string]interface{}{"title": title, "body": body})
	return article
}

File: src/service/article.js
import axios from 'axios';

class Article {
    async show(token) {
        try {
            const response = await axios.get('/api/bbs/show', { headers: { Authorization: token } })
            console.log(response)
            return response
        } catch (e) {
            console.log(e)
        }
    }
    async add(token, title, body) {
        try {
            await axios.post("/api/bbs/add", { title: title, body: body }, { headers: { "Authorization": token } })
        } catch (e) {
            console.log(e)
        }
    }
    async delete(token, id) {
        try {
            const response = await axios.post("/api/bbs/delete", { id: id }, { headers: { "Authorization": token } })
            response.message = "投稿を削除しました"
            return response
        } catch (e) {
            return {message: "他人の投稿は削除できません"}
        }
    }
    async edit(token, id, title, body) {
        try {
            const response = await axios.post("/api/bbs/edit", { id: id, title:title, body:body }, { headers: { "Authorization": token } })
            response.message = "投稿を編集しました"
            return response
        } catch (e) {
            return {message: "他人の投稿は編集できません"}
        }
    }
}

export default new Article();